### PR TITLE
[Documentation] Concepts: Countries

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/book/countries.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Provinces are added to a country via a collection. Create one and add the province object to it.
+    $provinces = new ArrayCollection();
+    $provinces->add($province);
+
+    // and using the prepared collection add the province to the Country.
+    $country->setProvinces($provinces);
+
+    // Finally you will need a repository for countries to add the country to your system.
+    /** @var EntityRepository $countryRepository */
```
Comment (pamil) [inline docs/book/countries.rst]: RepositoryInterface?

File: docs/book/countries.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /** @var ProvinceInterface $province */
+    $province = $provinceFactory->createNew();
+
+    // To the newly created objects assign proper codes.
+    // GB - Great Britain
+    $country->setCode('GB');
+    // GB_SCT - Scotland (which is a province of GB)
+    $province->setCode('GB_SCT');
+
+    // Provinces are added to a country via a collection. Create one and add the province object to it.
```
Comment (pamil) [inline docs/book/countries.rst]: `$country->addProvince($province);` should work as well

File: docs/book/countries.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Therefore we understand provinces as states of the USA, voivodeships of Poland, cantons of Belgium or bundesländer of Germany.
+
+How to add a country?
+---------------------
+
+To give you a better insight into Countries, let's have a look how to prepare and add a Country to the system programatically.
+We will do it together with a province at once.
+
+Firstly you will need factories for countries and for provinces. Create new objects with them.
+To the newly created objects assign codes. The provinces aree added to a country as a collection.
```
Comment (GSadee) [inline docs/book/countries.rst]: The provinces **are** added

File: docs/book/countries.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Finally using a repository for countries - add the country to the system.
+From now on it will be available to use.
+
+.. code-block:: php
+
+    <?php
+
+    // Firstly you will need factories for countries and for provinces:
+    /** @var FactoryInterface $countryFactory **/
+    $countryFactory= $this->get('sylius.factory.country');
```
Comment (GSadee) [inline docs/book/countries.rst]: Missing space before `=`

File: docs/book/countries.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+Countries
+=========
+
+**Countries** are a part of the :doc:`Addressing </book/addresses>` concept.
+The **Country** entity represents a real country that your shop is willing to sell its goods in (for example the UK).
+It has an ISO code to be identified easily (ISO 3166-1 alpha-2).
+
+Countries might also have **Provinces**, which is in fact a general name for an administrative division, within a country.
+Therefore we understand provinces as states of the USA, voivodeships of Poland, cantons of Belgium or bundesländer of Germany.
```
Comment (pjedrzejewski) [inline docs/book/countries.rst]: 👍 👍 👍

**Comment by pjedrzejewski** [comment]
Maybe we can choose a bit less controversial example of provinces. :D

**Comment by TheMadeleine** [comment]
Done :D We've got California now. 🌴

**Comment by pjedrzejewski** [comment]
Thank you Magda!

**Pull request merged**